content/3_Silver/Prefix_Sums_2.mdx

```diff
@@ -102,11 +102,98 @@ export const problems = {
 
 <FocusProblem problem={problems.maxsum} />
 
-This problem has a solution known as [Kadane's Algorithm](https://en.wikipedia.org/wiki/Maximum_subarray_problem#Kadane's_algorithm). Please don't use that solution; try to solve it with prefix sums.
+This problem has two types of solutions one of which is called the  [Kadane's Algorithm](https://en.wikipedia.org/wiki/Maximum_subarray_problem#Kadane's_algorithm). However, understanding Kadane's algorithm is **optional** for this problem, since it can also be solved using prefix sums.
 
-<Spoiler title="Why are the two methods equivalent?">
+<Spoiler title="Solution">
+
+### Solution
+
+Suppose that we store the prefix sums at each index in the array. Intuitively, the maximum subarray sum ending at each index is the difference between the maximum prefix sum and the minimum prefix sum. Thus, we can iterate each element in the array and store the minimum prefix sum up to the current index. Then, we can find the maximum subarray sum by subtracting the current prefix sum to the minimum prefix sum.
+
+<LanguageSection>
+
+<CPPSection>
+
+C++ code:
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+using ll = long long;
+using vl = vector<ll>;
+
+int main(){
+	int n; cin >> n;
+	vl dp(n+1, 0); // Prefix sum array initially filled with 0's
+	for(int i=1; i<=n; i++){
+		ll x; cin >> x;
+		dp[i] = dp[i-1] + x; // Compute the prefix sum at each element
+	}
+	ll max_subarray_sum = dp[0], min_prefix_sum = dp[0];
+	for(int i=1; i<=n; i++){
+		// The max subarray sum is the maximum difference between two prefix sums
+		max_subarray_sum = max(max_subarray_sum, dp[i] - min_prefix_sum);
+		min_prefix_sum = min(min_prefix_sum, dp[i]);
+	}
+	cout << max_subarray_sum << endl;
+}
+```
+
+</CPPSection>
+
+</LanguageSection>
+
+</Spoiler>
+
+<Warning>
+
+Note: It is recommended that you skip the following solution.
+
+</Warning>
+
+<Spoiler title = "Alternative Solution">
+
+### Solution
+
+Suppose we fix the first element of a subarray. The Kadane's Algorithm solution involves starting a new subarray sum at an element if that element is greater than the current subarray sum with an addition to the current element. If the subarray sum ending at the current element is greater than the current element, then we continue the subarray sum. Otherwise, we set the new subarray sum equal to the current element and restart our subarray sum. To find the maximum subarray sum over all possible sums, we store an integer $S$, representing the maximum subarray sum at each iteration of the array. When we finish iterating the array, we print $S$.
```

Comment:
Not very fluid.

"if that element is greater than the current subarray sum with an addition to the current element" What does this mean?

"continue the subarray sum" what does this mean? Extend the current subarray to the right?

" To find the maximum subarray sum over all possible sums, we store an integer $S$, representing the maximum subarray sum at each iteration of the array. When we finish iterating the array, we print $S$." is the easy part. You can just simplify it to "print the largest subarray sum obtained by the algorithm" or something

Reply:
Alternatively you can entirely waive the details by saying like "Kadane's Algorithm finds the largest subarray sum using a greedy method. Check the video below for details"